### ISSUE TITLE
Incorrect error message when a !-declared variable has an initializer

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#40663]
Kind of a late follow-up to https://github.com/microsoft/TypeScript/issues/30697

```ts
let x!: string = "hello";
```

**Current**

```
Definite assignment assertions can only be used along with a type annotation.
```

**Proposed**

```
Declarations with initializers cannot also have definite assignment assertions.
```

Note that any improvement here **should not** regress the error messages for other tests.
